modules/generative-openai/additional/generate/generate_result.go

```diff
@@ -0,0 +1,146 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2023 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package generate
+
+import (
+	"context"
+	"regexp"
+	"sync"
+
+	"github.com/pkg/errors"
+	"github.com/weaviate/weaviate/entities/models"
+	"github.com/weaviate/weaviate/entities/moduletools"
+	"github.com/weaviate/weaviate/entities/search"
+	generativemodels "github.com/weaviate/weaviate/modules/generative-openai/additional/models"
+	"github.com/weaviate/weaviate/modules/generative-openai/ent"
+)
+
+var maximumNumberOfGoroutines = 10
```

Comment:
I would move this setting to `generete.go` and make it a const.
I would also define this number inside the `GenerateProvider` struct and use it from it, so that it would be more visible that we have it defined inside the provider
```
type GenerateProvider struct {
	client openAIClient
        maximumNumberOfGoroutines int
}
```